src/Tribe/Views/V2/View.php

```diff
@@ -2194,50 +2194,79 @@ protected function should_show_recent_past_events_view() {
 	 */
 	protected function setup_additional_views( array $events = [], array $template_vars = [] ) {
 
-		$manager = tribe( Manager::class );
+		$manager      = tribe( Manager::class );
 		$default_slug = $manager->get_default_view_option();
 
-		// Only recent Past Events to only the default view.
+		// Show Latest Past Events only on the default view.
 		if ( $this->get_slug() !== $default_slug ) {
 			return;
 		}
 
-		// Flatten Month and Week View Events to Get Accurate Count
-/*		$views_to_flatten = [
-			'month',
-			'week',
-		];
+		$now    = $this->context->get( 'now', time() );
+		$latest = tribe_events_latest_date();
+
+		// If now is less then the latest event published, do not show.
+		if ( $now < $latest ) {
+			return;
+		}
+
+		// Checks to verify on the initial load of a view or if using today's date for the view.
+		$today     = $this->context->get( 'today' );
+		$view_date = $this->context->get( 'event_date', '' );
+
+		switch ( $this->get_slug() ) {
+			case 'month':
+				$today_formatted     = Dates::build_date_object( $today )->format( Dates::DBYEARMONTHTIMEFORMAT );
+				$view_date_formatted = Dates::build_date_object( $view_date )->format( Dates::DBYEARMONTHTIMEFORMAT );
+				break;
+
+			case 'week':
+				list( $today_week_start, $today_week_end ) = Dates::get_week_start_end( $today, (int) $this->context->get( 'start_of_week', 0 ) );
+				list( $view_week_start, $view_week_end )   = Dates::get_week_start_end( $view_date, (int) $this->context->get( 'start_of_week', 0 ) );
 
-		if ( in_array( $this->get_slug(), $views_to_flatten ) ) {
+				$today_formatted     = $today_week_start->format( Dates::DBDATEFORMAT );
+				$view_date_formatted = $view_week_start->format( Dates::DBDATEFORMAT );
+				break;
+
+			default:
+				$today_formatted     = Dates::build_date_object( $today )->format( Dates::DBDATEFORMAT );
+				$view_date_formatted = Dates::build_date_object( $view_date )->format( Dates::DBDATEFORMAT );
+		}
+
+		// If view date is not empty and today does not equal the view date, then do not show.
+		if ( ! empty( $view_date ) && $today_formatted !== $view_date_formatted ) {
+			return;
+		}
+
+		// Flatten Views such as Month and Week that have an array values.
+		$first_value = reset($events);
```

Comment:
* :no_entry_sign: Expected 1 spaces after opening bracket; 0 found
* :no_entry_sign: Expected 1 spaces before closing bracket; 0 found
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis

<sub>hash: da0589281994d316b1ab2080197a839c</sub>